questionGenerator.py

```diff
@@ -0,0 +1,318 @@
+import pandas as pd
+from utils import dataLoader, dataSaver, printAndLog
+from config import OpenAIBot
+import json
+import sys
+
+
+class QuestionData:
+    """
+    Represents a class that handles question data generation.
+
+    Args:
+        config (Config): The configuration object.
+        videoData (VideoData): The video data object.
+        topicModeller (TopicModeller): The topic modeller object.
+        load (bool, optional): Whether to load existing question data. Defaults to True.
+    """
+
+    def __init__(self, config, videoData, topicModeller, load=True):
```

Comment:
I would avoid writing the huge business logic when instantiate the class.  Like loading and saving data, clustering , filtering dominats topic and instantiating OpenAIBot. That's a lot.

Create a function like some kind of orchestrator  after instantiate the class that handle all the workflow going with Question genration